src/Sylius/Component/Core/Distributor/MinimumPriceDistributor.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Distributor;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class MinimumPriceDistributor implements MinimumPriceDistributorInterface
+{
+    private ProportionalIntegerDistributorInterface $proportionalIntegerDistributor;
+
+    public function __construct(ProportionalIntegerDistributorInterface $proportionalIntegerDistributor)
+    {
+        $this->proportionalIntegerDistributor = $proportionalIntegerDistributor;
+    }
+
+    public function distribute(array $orderItems, int $amount, ChannelInterface $channel): array
+    {
+        $orderItemsToProcess = [];
```

Comment:
```suggestion
        Assert::allInstanceOf(OrderItemInterface::class, $orderItems);
        
        $orderItemsToProcess = [];
```

Reply:
i implemented Assert::allImplementsInterface($orderItems, OrderItemInterface::class); as it couldnt find method allInstanceOf :/